middleware/kubernetes/federation.go

```diff
@@ -0,0 +1,76 @@
+package kubernetes
+
+import (
+	"strings"
+
+	"github.com/coredns/coredns/middleware/etcd/msg"
+)
+
+type Federation struct {
+	name string
+	zone string
+}
+
+const (
+	// TODO: Do not hardcode these labels. Pull them out of the API instead.
+	//
+	// We can get them via ....
+	//   import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//     metav1.LabelZoneFailureDomain
+	//     metav1.LabelZoneRegion
+	//
+	// But importing above breaks coredns with flag collision of 'log_dir'
```

Comment:
we don't have `log_dir`, maybe glog does? We already remove `log_dir` from coredns' flags.